pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -800,6 +793,36 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(podInfoList []*framework.
 	}
 }
 
+func (p *PriorityQueue) movePodToActiveOrBackoffQueue(pInfo *framework.QueuedPodInfo, avoidBackoff bool, event framework.ClusterEvent) bool {
+	// If the event doesn't help making the Pod schedulable, then keep it in its current state.
+	// Note: we don't run the check if pInfo.UnschedulablePlugins is nil, which denotes
+	// either there is some abnormal error, or scheduling the pod failed by plugins other than PreFilter, Filter and Permit.
+	// In that case, it's desired to move it anyways.
+	if len(pInfo.UnschedulablePlugins) != 0 && !p.podMatchesEvent(pInfo, event) {
+		return false
+	}
+
+	pod := pInfo.Pod
+	if !avoidBackoff && p.isPodBackingoff(pInfo) {
+		if err := p.podBackoffQ.Add(pInfo); err != nil {
+			klog.ErrorS(err, "Error adding pod to the backoff queue", "pod", klog.KObj(pod))
+		} else {
+			klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", event, "queue", backoffQName)
+			metrics.SchedulerQueueIncomingPods.WithLabelValues("backoff", event.Label).Inc()
+			p.unschedulablePods.delete(pod, pInfo.Gated)
+		}
+	} else {
```

Comment:
Is it OK to ignore backoff?

What if plugin returns PodImmediatelySchedulable to many Pods and many Pods goes to activeQ without backoff? Doesn't that affect badly to other Pod's scheduling?

Reply:
The way I understand "backoff" is that something *unexpected* has happened, like for example a cluster event might have been missed or some internal error.

The problem is that some features, for example DRA (tested) and "wait for first consumer" volume provisioning (not tested as far as I know), by design need multiple scheduling attempts and currently enter backoff during a *normal* flow of events. If we don't avoid that, scheduling of pods using such features will always be slow.

I think it's desirable to avoid backoff for those events where "this is normal" could be detected.

Reply:
Yeah, it won't be a concern if it's accurate enough though. 

@pohly 
I don't have much insight on DRA impl yet, but how accurately could the dynamicresource plugin return `PodImmediatelySchedulable`? 
Will a Pod almost 100% get successfully scheduled in next scheduling cycle when it gets `PodImmediatelySchedulable`? or Is that very depends on other scheduler plugin's result as well?

---

@alculquicondor @Huang-Wei It may be a very basic question, but what's the reason we have backoffQ?

Reply:
Backoff is for scheduling fairness; otherwise keep trying high-priority pods would cause a HOL issue.

Reply:
OK, understood. 
So, `PodImmediatelySchedulable` is for only very limited cases where the Pod will very highly likely get scheduled in next scheduling cycle like dyamice resource plugin. 
And, like the usecase of #114297, we cannot 100% say "this Pod should be scheduled in next scheduling cycle", and thus we should use PodMaybeSchedulable.